src/util.rs

```diff
@@ -6,22 +6,62 @@ use std::str;
 
 use memchr::{memchr, memchr2};
 
-#[cfg(feature = "compression")]
-use zip::result::ZipError;
-
 #[derive(Clone, Debug, PartialEq)]
-pub enum ParseError {
+pub enum ParseErrorType {
+    BadCompression,
     PrematureEOF,
-    Invalid(String),
+    InvalidHeader,
+    InvalidRecord,
+    IOError,
+    Invalid,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct ParseError {
+    pub record: usize,
```

Comment:
could the line be tracked as well? The record number doesn't make it easy to check for fasta files